src/main/java/com/example/saaca/syncup/controller/ReturnFormController.java

```diff
@@ -57,4 +69,21 @@ public int deleteReturnForms(@PathVariable(value = "returnType")final String ret
         return returnFormRepository.findAll();
     }
 
+    @PutMapping("/revised-due-date/{form_name}")
+    public ReturnForm addRevisedDueDateOfFiling(@PathVariable(value = "form_name")final String formName,
+                                             @RequestBody final DueDateScheduler dueDateScheduler) {
+        List<ReturnForm> returnForms = returnFormRepository.findByFormNames(Arrays.asList(formName));
+        if (returnForms == null) {
+            return null;
+        }
+        ReturnForm returnForm = returnForms.get(0);
+        Date currentDate = new Date();
+        Date revisedDueDate = dueDateScheduler.getRevisedDueDateOfFiling();
+        for (DueDateScheduler scheduler: returnForm.getDueDateSchedulerSet()) {
+            if (scheduler.getToBeDelete() == 0) {
+                scheduler.setRevisedDueDateOfFiling(revisedDueDate);
```

Comment:
this is sort of edit as well ? 
if a revised date already present then it will be replaced, have we validated that scenario ?

Reply:
yes it will be replaced and that is to be done